packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -106,14 +136,22 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
 
   @override
   void addAllowedPointer(PointerEvent event) {
-    // Ignore out-of-bounds second taps.
-    if (_firstTap != null &&
-        !_firstTap.isWithinTolerance(event, kDoubleTapSlop))
-      return;
+    if (_firstTap != null) {
+      // Ignore out-of-bounds second taps.
+      if (!_firstTap.isWithinTolerance(event, kDoubleTapSlop)) {
+        return;
+      }
+      // Restart when the second tap is too close to the first
```

Comment:
I think I'd move this comment up to the top of the if block, and keep the `else if` on the same line as the `}`

Reply:
I understand the idea of keeping `else` next to `if`. What do you think if we move the comments of both branch to the top within the branch, like
```
if () {
  // comment
} else if () {
  // comment
}
```

Reply:
That's fine too.